docs/configure-custom-values-envoy.md

```diff
@@ -161,6 +134,21 @@ envoy:
     allowPrivilegeEscalation: false
 ```
 
+### Image configurations (Default value is recommended)
+
+If you want to change the image repository and version, you can use `envoy.image.repository`. Please specify the container repository information that you pull the Scalar Envoy container image.
```

Comment:
I think `be sure to` is not appropriate in case of Scalar Envoy. In the Scalar Envoy document, we have to make the `image.repository` configuration an optional thing (not mandatory).

Whether the `image.repository` configuration is mandatory or not depends on whether the image is public or private.

---

## In case of not mandatory

For example, Scalar Envoy container images are published in the `public` container registry.
https://github.com/orgs/scalar-labs/packages/container/package/scalar-envoy

So, users can pull it without any subscribing or authentication. In this case, users can pull images with default values.

Therefore, the `image.repository` configuration is NOT mandatory.

---

## In case of mandatory

On the other hand, for example, ScalarDL images are published in the `private` container registry (private GitHub Packages).

So, basically, users have to pull those images from AWS/Azure Marketplace instead of private GitHub Packages.

In other words, the default value of the `image.repository` configuration is set to the private GitHub Packages.

Note: We set the private GitHub Packages to the `image.repository` configuration as a default value. This is because we (Scalar developers) use the private GitHub Packages and we don't have an appropriate default value (default container registry) for users. This configuration depends on which cloud service provider the user chooses.

Therefore, the `image.repository` configuration is mandatory.

---

In conclusion, whether the `image.repository` configuration is mandatory or not is as follows.

* Not mandatory (published in the public container registry)
  * Scalar Envoy
  * ScalarDB Analytics for PostgreSQL (in the future)
* Mandatory (published in the private container registry with commercial license)
  * ScalarDB Server
    * For historical reasons, ScalarDB Server images exist on the public registry, but we treat it as a private (commercial license) thing.
  * ScalarDB Cluster
  * ScalarDL Ledger
  * ScalarDL Auditor
  * ScalarDL Schema Loader
  * Scalar Manager
  * Other commercial license products (in the future)

Reply:
I see. Thank you for the explanation!

In that case, how about the following?:

```suggestion
If you want to change the image repository and version, you can use `envoy.image.repository` to specify the container repository information of the Scalar Envoy container image that you want to pull.
```

Reply:
Thank you for your suggestion!
Looks good!
I will apply this suggestion.